### PR TITLE
Initial Nest binding.

### PR DESCRIPTION
**Pull request by watou**
Pull request to add Nest binding to openHAB.

### CODE REVIEW

**Comment by watou** [comment]
The wiki is [here](https://github.com/openhab/openhab/wiki/Nest-Binding).

**Comment by teichsta** [comment]
Thanks @watou for this contribution. Hope to find the time for the Review soon! Thanks, Thomas E.-E.

File: bundles/binding/org.openhab.binding.nest/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+Bundle-SymbolicName: org.openhab.binding.nest
+Bundle-Vendor: openHAB.org
+Bundle-Version: 1.7.0.qualifier
+Bundle-Activator: org.openhab.binding.nest.internal.NestActivator
+Bundle-ManifestVersion: 2
+Bundle-Description: This is the Nest binding of the open Home Aut
+ omation Bus (openHAB)
+Import-Package: com.google.common.base,
+ com.google.common.io,
+ org.apache.commons.httpclient,
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.nest/META-INF/MANIFEST.MF]: either add version (3.1.0) as well oder remove from all packages

File: bundles/binding/org.openhab.binding.nest/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ org.apache.commons.httpclient.auth;version="3.1.0",
+ org.apache.commons.httpclient.methods;version="3.1.0",
+ org.apache.commons.httpclient.params;version="3.1.0",
+ org.apache.commons.httpclient.util,
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.nest/META-INF/MANIFEST.MF]: either add version (3.1.0) as well oder remove from all packages

File: bundles/binding/org.openhab.binding.nest/META-INF/MANIFEST.MF [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+ org.apache.commons.io;version="2.0.1",
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.nest/META-INF/MANIFEST.MF]: are you sure, the version is neccessary?

File: bundles/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/NestBinding.java [hunk 4]
```diff
@@ -0,0 +1,588 @@
...
+		if (command instanceof State) {
+			updateNest(itemName, (State) command);
+		}
+	}
+
+	private boolean isEcho(String itemName, State state) {
+		String ignoreEventSetKey = itemName + state.toString();
+		if (ignoreEventSet.contains(ignoreEventSetKey)) {
+			ignoreEventSet.remove(ignoreEventSetKey);
+			logger.trace(
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/NestBinding.java]: increase to Debug-Level

**Comment by teichsta** [comment]
@watou thanks again for this great Binding. See some (minor) comments inline. Great code quality, great work! Looking forward to many new Bindings from you :-)

**Comment by watou** [comment]
Thank you, Thomas, for taking the time to review it!

**Comment by teichsta** [comment]
@watou it would be great if you could talk/tweet/nameit about this new Binding. Since Nest gained "some" traction in the past it would be great for openHAB also. Also great would be a short Video showing the interaction of openHAB and Nest (see examples in the openHAB Youtube Channel - https://www.youtube.com/channel/UC7OK88DW0La_BJlcXZg8ydQ/videos) …

**Comment by teichsta** [comment]
@watou see my comment above would you be able to create some short video, screencast, photo … anything that will help @kaikreuzer to pimp the press release for the upcoming release on Monday? That would be REALLY great :-) Thanks, Thomas E.-E.

**Comment by jensihnow** [comment]
No Nest here, but nest protects: Already posted something https://twitter.com/JIhnow/status/576877606735347712

**Pull request merged**